database-commons/src/main/java/io/cdap/plugin/util/DBUtils.java

```diff
@@ -184,6 +189,60 @@ public static void validateJDBCPluginPipeline(PipelineConfigurer pipelineConfigu
       jdbcPluginId, PluginProperties.builder().build());
   }
 
+  /**
+   * Checks if fields from schema are compatible to be written into database.
+   *
+   * @param actualSchema schema from db.
+   * @param configSchema schema from config.
+   */
+  public static void validateSourceSchema(Schema actualSchema, Schema configSchema) {
+    if (configSchema == null) {
+      throw new InvalidConfigPropertyException("Schema should not be null or empty",
+                                               AbstractDBSource.DBSourceConfig.SCHEMA);
```

Comment:
this validate method is only ever correct to be called from a source. It's better to keep it in the AbstractDBSource, but make it a package private static method that you can call from unit tests.

Reply:
done